pkg/build/registry/buildconfig/strategy.go

```diff
@@ -39,16 +52,17 @@ func (strategy) AllowUnconditionalUpdate() bool {
 	return false
 }
 
-// PrepareForCreate clears fields that are not allowed to be set by end users on creation.
-func (strategy) PrepareForCreate(ctx apirequest.Context, obj runtime.Object) {
-	bc := obj.(*api.BuildConfig)
-	dropUnknownTriggers(bc)
-}
-
 // Canonicalize normalizes the object after validation.
 func (strategy) Canonicalize(obj runtime.Object) {
 }
 
+// PrepareForCreateCommon clears fields that are not allowed to be set by end users on creation.
+// This is shared by the Group and Legacy strategies.
+func (s strategy) PrepareForCreateCommon(ctx apirequest.Context, obj runtime.Object) {
```

Comment:
if i didn't want to share the common behavior, yes.  but I didn't want to write the common behavior in both impls.  Am I missing a composition pattern?